src/init/starship.xsh

```diff
@@ -0,0 +1,11 @@
+def starship_prompt():
+    last_cmd = __xonsh__.history[-1] if __xonsh__.history else None
+    status = last_cmd.rtn if last_cmd else 0
+    jobs = len(__xonsh__.all_jobs)
+    duration = round((last_cmd.ts[1] - last_cmd.ts[0]) * 1000) if last_cmd else 0
+    return $(::STARSHIP:: prompt --status=@(status) --jobs=@(jobs) --cmd-duration=@(duration))
+
+
+$PROMPT = starship_prompt
+$STARSHIP_SHELL = "xonsh"
+$STARSHIP_SESSION_KEY = $(::STARSHIP:: session).strip()
```

Comment:
Is it possible to generate this from `xonsh` instead? Something like `"".join(secrets.choice(string.ascii_letters + string.digits) for _ in range(16))`?

Reply:
Done. I used `random` rather than `secrets` because xonsh's minimum Python version is 3.5, which did not yet have `secrets`. I assume this key does not need cryptographically secure randomness, but please let me know if I'm mistaken about that.